packages/visualizations/addon/components/navi-visualizations/table.js

```diff
@@ -201,12 +203,20 @@ export default Component.extend({
    */
   columns: computed('options.columns', 'request.sort', function() {
     let sorts = this._mapAlias(get(this, 'request')),
-        columns = $.extend(true, [], get(this, 'options.columns'));
+        columns = cloneDeep(get(this, 'options.columns') || []),
+        columnIndex = get(this, 'columnIndex');
 
-    return columns.map( column => {
+    if (!columnIndex) {
+      columnIndex = indexColumnById(columns);
+      set(this, 'columnIndex', columnIndex);
+    }
+
+    return columns.map(column => {
       let {field, type} = column,
           fieldName = type === 'dateTime' ? type : canonicalizeMetric(field),
           sort =  arr(sorts).findBy('metric', fieldName) || {},
+          oldColumn = columnIndex[column.field[column.type === 'threshold' ? 'metric' : column.type]],
+          isEdited = !isEqual(column, oldColumn),
```

Comment:
is this `hasCustomDisplayName`?

Reply:
I guess we also want to show edited state when format changed

Reply:
Should we compare the display name to the default display name?

Reply:
I think it should be compared to the initial display name